### ISSUE TITLE
Complex, multiline, multi-lambda expressions in ampclient.py example make it more difficult than necessary to follow

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6204]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#6204|
|Type|defect|
|Created|2012-12-06 13:15:32Z|
|Branch|https://github.com/twisted/twisted/tree/simpler-ampclient-example-6204|

We all know it is _possible_ to write code like:

```python
    d1 = ClientCreator(reactor, amp.AMP).connectTCP(
        '127.0.0.1', 1234).addCallback(
            lambda p: p.callRemote(Sum, a=13, b=81)).addCallback(
                lambda result: result['total'])
```

And for many people, it's even possible to read such code.  However, many more people have an easier time reading something where each step is separate and where lambda is not used to define a function inside a call to `addCallback`.  Particularly since the reader of any of these examples is already trying to learn how to use _AMP_, they don't need to be given the extra burden of untangling this less-than-perfectly readable style.

<details><summary>Searchable metadata</summary>

```
trac-id__6204 6204
type__defect defect
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__branches_simpler_ampclient_example_6204 branches/simpler-ampclient-example-6204
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__core core
keywords__documentation documentation
time__1354799732000000 1354799732000000
changetime__1354891130000000 1354891130000000
version__None None
owner__exarkun exarkun

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

(In [36525]) Branching to 'simpler-ampclient-example-6204'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun removed owner|
|-|-|

I think that's better.  There are still problems with the example, such as having no explanation about how to use it or what behavior to expect from it, or the continuing use of `ClientCreator` instead of endpoints, or manual reactor management.  I think those would all make good tickets _separate_ from this one, which makes one well-defined improvement to the readability of the example.

**Comment by twisted-trac** [comment]
|[<img alt="teratorn's avatar" src="https://avatars.githubusercontent.com/u/154069?s=50" width="50" height="50">](https://github.com/teratorn)|@teratorn set owner to @teratorn|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="teratorn's avatar" src="https://avatars.githubusercontent.com/u/154069?s=50" width="50" height="50">](https://github.com/teratorn)<a name="note_4"></a>|@teratorn set owner to @exarkun|
|-|-|

I've reviewed the proposed change to ampclient.py and I like the new style. It's worth noting the "inline" placement of adding Deferred callbacks. Haven't seen that much - and it does seem to help with readability.

I guess we don't have any test case policy for example code :/ (it does appear to work, still)

OK for .misc entry, but if any other improvements take place we should probably mention the improved AMP examples in a news file.

Please merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun set status to `closed`|
|-|-|

(In [36534]) Merge simpler-ampclient-example-6204 

Author: exarkun
Reviewer: teratorn
Fixes: [#6204](https://github.com/twisted/twisted/issues/6204)

Simplify the Python style used in the AMP client example.
